From 08e817e91f44f947f8945115302100500b51ae22 Mon Sep 17 00:00:00 2001
Message-Id: <08e817e91f44f947f8945115302100500b51ae22.1366126535.git.jtomko@redhat.com>
From: =?UTF-8?q?J=C3=A1n=20Tomko?= <jtomko@redhat.com>
Date: Fri, 12 Apr 2013 17:30:56 +0200
Subject: [PATCH] daemon: fix leak after listing all volumes
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

remoteDispatchStoragePoolListAllVolumes wasn't freeing the pool.
The pool also held a reference to the connection, preventing it from
getting freed and closing the netcf interface driver, which held two
sockets open.
---
 daemon/remote.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/daemon/remote.c b/daemon/remote.c
index 45c50f3..37f998f 100644
--- a/daemon/remote.c
+++ b/daemon/remote.c
@@ -4229,6 +4229,8 @@ cleanup:
             virStorageVolFree(vols[i]);
         VIR_FREE(vols);
     }
+    if (pool)
+        virStoragePoolFree(pool);
     return rv;
 }
 
-- 
1.8.1.5

